scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java

```diff
@@ -771,10 +803,54 @@ private void dispatchJobStateUpdated(String owner, NotificationData<JobInfo> not
     }
 
     /**
+     * Dispatch the job state updated event
+     * 
+     * @param owner
+     *            the owner of this job
+     * @param notification
+     *            the data to send to every client
+     */
+    private void dispatchJobUpdatedFullData(JobState job) {
+        try {
+            if (logger.isDebugEnabled()) {
+                jlogger.debug(job.getJobInfo().getJobId(), " event [" + SchedulerEvent.JOB_UPDATED + "]");
+            }
+            for (ListeningUser listeningUserId : identifications.values()) {
+                // if this user has a listener
+                if (listeningUserId.isListening()) {
+                    try {
+                        UserIdentificationImpl userId = listeningUserId.getUser();
+                        // if there is no specified event OR if the specified
+                        // event is allowed
+                        if ((userId.getUserEvents() == null) ||
+                            userId.getUserEvents().contains(SchedulerEvent.JOB_UPDATED)) {
+                            // if this userId have the myEventOnly=false or
+                            // (myEventOnly=true and it is its event)
+                            if (!userId.isMyEventsOnly() ||
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1066)